x-pack/legacy/plugins/lens/public/indexpattern_plugin/dimension_panel/dimension_panel.tsx

```diff
@@ -14,6 +14,8 @@ import {
   SavedObjectsClientContract,
   HttpServiceBase,
 } from 'src/core/public';
+import { npStart } from 'ui/new_platform';
```

Comment:
We have a rule that we don't import new platform code from anything that isn't at the top level, because it won't be possible in the future. Most of the properties you need are already passed in to the `getIndexPatternDatasource` function from `indexpattern_plugin/plugin.tsx`, but not the data plugin.

So I suggest you move this `KibanaContextProvider` to the `renderDimensionPanel` function in `getIndexPatternDatasource`, and also pass in the `data` plugin from `indexpattern_plugin/plugin.tsx`